pkg/grpc/api_crypto_test.go

```diff
@@ -0,0 +1,565 @@
+				&runtimev1pb.DecryptAlpha1Request{
+					Options: &runtimev1pb.DecryptAlpha1RequestOptions{
+						ComponentName: "myvault",
+					},
+				},
+				&runtimev1pb.DecryptAlpha1Request{
+					Payload: &commonv1pb.StreamPayload{
+						Seq:  0,
+						Data: enc,
+					},
+				},
+			}
+			dec, err := cryptoSendRequest(stream, send, &runtimev1pb.DecryptAlpha1Response{})
+			require.NoError(t, err)
+			require.Equal(t, "hello world", string(dec))
+		})
+	})
+
+	t.Run("multiple data chunks", func(t *testing.T) {
+		var enc []byte
+		t.Run("encrypt", func(t *testing.T) {
+			stream, err := client.EncryptAlpha1(context.Background())
+			require.NoError(t, err)
+			send := []runtimev1pb.CryptoRequests{
+				&runtimev1pb.EncryptAlpha1Request{
+					Options: &runtimev1pb.EncryptAlpha1RequestOptions{
+						ComponentName: "myvault",
+						KeyName:       "aes-passthrough",
+						Algorithm:     "AES",
+					},
+				},
+				&runtimev1pb.EncryptAlpha1Request{
+					Payload: &commonv1pb.StreamPayload{
+						Seq:  0,
+						Data: []byte("soft kitty, warm kitty, little ball of fur, "),
+					},
+				},
+				&runtimev1pb.EncryptAlpha1Request{
+					Payload: &commonv1pb.StreamPayload{
+						Seq:  1,
+						Data: []byte("happy kitty, sleepy kitty, purr purr purr"), //nolint:dupword
+					},
+				},
+			}
+			enc, err = cryptoSendRequest(stream, send, &runtimev1pb.EncryptAlpha1Response{})
+			require.NoError(t, err)
+			require.True(t, bytes.HasPrefix(enc, []byte("dapr.io/enc/v1")))
+		})
+
+		t.Run("decrypt - whole header in first chunk", func(t *testing.T) {
+			stream, err := client.DecryptAlpha1(context.Background())
+			require.NoError(t, err)
+			send := []runtimev1pb.CryptoRequests{
+				&runtimev1pb.DecryptAlpha1Request{
+					Options: &runtimev1pb.DecryptAlpha1RequestOptions{
+						ComponentName: "myvault",
+					},
+				},
+				&runtimev1pb.DecryptAlpha1Request{
+					Payload: &commonv1pb.StreamPayload{
+						Seq:  0,
+						Data: enc[0:180],
+					},
+				},
+				&runtimev1pb.DecryptAlpha1Request{
+					Payload: &commonv1pb.StreamPayload{
+						Seq:  1,
+						Data: enc[180:],
+					},
+				},
+			}
+			dec, err := cryptoSendRequest(stream, send, &runtimev1pb.DecryptAlpha1Response{})
+			require.NoError(t, err)
+			require.Equal(t, "soft kitty, warm kitty, little ball of fur, happy kitty, sleepy kitty, purr purr purr", string(dec)) //nolint:dupword
+		})
+
+		t.Run("decrypt - header split in multiple chunks", func(t *testing.T) {
+			stream, err := client.DecryptAlpha1(context.Background())
+			require.NoError(t, err)
+			send := []runtimev1pb.CryptoRequests{
+				&runtimev1pb.DecryptAlpha1Request{
+					Options: &runtimev1pb.DecryptAlpha1RequestOptions{
+						ComponentName: "myvault",
+					},
+					Payload: &commonv1pb.StreamPayload{
+						Seq:  0,
+						Data: enc[0:50],
+					},
+				},
+				&runtimev1pb.DecryptAlpha1Request{
+					Payload: &commonv1pb.StreamPayload{
+						Seq:  1,
+						Data: enc[50:180],
```

Comment:
probably define header size and then set the magic numbers as vars or constants :)

Reply:
See above. Also note header size isn't constant.